### DISCUSSION TITLE
Nova extension

### DISCUSSION THREAD

**Discussion by joshbruce** [community/community#7431] [Copilot Conversations]
I currently use two editors:

1. https://www.nova.app and 
2. https://www.sublimetext.com/download

Currently there doesn't seem to be a copilot extension available for either one.

Curious if they are coming or how to possibly build an extension for the community??

Sorry if this is a repeat as I keep seeing and wanting to use copilot - just don't want to switch editors to do it.

**Comment by asowers1** [comment]
+1 for Nova

**Comment by cobyism** [comment]
I volunteer to help write this extension if others want to help me, I believe my experience at GitHub and Intercom, and using GPT-3 and other ML tooling myself makes me qualified to do so (or so I hope lol).

Anyone else out there building this already?

**Reply by michaelaflores** [reply]
Happy to help as well – did this ever get started? It appears there's a way as it was added for Sublime https://github.com/TheSecEng/LSP-copilot

**Comment by piotrzaborow** [comment]
If someone from GitHub will just help us with docs to the Copilot service, we might write it ourselves. I am all in to help with the extension for Nova.

**Comment by eivindml** [comment]
Would love to see some support, especially now it's out of beta.

**Comment by iansvo** [comment]
I would be interested in Nova support and could be interested in contributing to it, but this kind of thing isn't super familiar to me so I'd want to work with others on it.

**Comment by BrandonKnudsen** [comment]
Can someone start a project and link to it? We all want to help.

**Comment by zerolfx** [comment]
For Sublime users: https://github.com/TheSecEng/LSP-copilot

**Comment by SkelegonDK** [comment]
+1 for Nova
I literally changed back to VS Code because of copilot. 
I'm not an expert in making extensions but I'd love to help in any way I can.

**Comment by rileyrichter** [comment]
Almost a year later and still no extension. 😭

**Comment by little-flag** [comment]
I agree that a Nova extension would be extremely desirable!

**Comment by gabltz** [comment]
since no Nova extension ?

**Comment by billh93** [comment]
+1 for Nova extension

**Comment by carlogatto** [comment]
+1 Nova extension

**Comment by fbracht** [comment]
+1 for Nova.

Based on the fact that the frequency of these comments here is increasing almost exponentially, I wonder what's been the actual holdup.

**Comment by joshbruce** [comment]
To give an update from the person who originally posted this.

I've actually stopped pursuing looking into copilot. 

For what it's worth, I also don't use Sublime Text at the moment. I'm appreciating where Nova is going as a native macOS experience. However, I know there are a fair amount of Sublime Text users.

**Comment by fhauck** [comment]
+1 for Nova 🙌

**Comment by avasilic** [comment]
+1 for nova

**Comment by oliwoodsuk** [comment]
+1 for Nova. It's just leaving a huge wad of productivity on the table by not having it. Currently, I'm using ChatGTP for little bits when I can, but it definitely not ideal.

**Comment by randomsnowflake** [comment]
Another +1 for Nova

**Comment by Antoineview** [comment]
+1 For nova !

**Comment by CodeyGuyDylan** [comment]
+1 for nova

**Reply by Javier-G-Lainez** [reply]
+1 for Nova

**Comment by jwilcoxson** [comment]
+1 on Nova

**Comment by nathanielperales** [comment]
+1 on Nova

**Comment by kitsah** [comment]
+100 for Nova!

**Comment by ErikVeland** [comment]
+1000 for Nova!

**Comment by fireboxcreative** [comment]
+1 Nova!

**Comment by pgorrindo** [comment]
+1 for Nova!

**Comment by Halopend** [comment]
+1 on Nova!

**Comment by andreanepori** [comment]
+1 on nova!

**Comment by TheBoatyMcBoatFace** [comment]
It looks like we may be getting a step closer...


I haven't played around with building Nova extensions, but it should be a somewhat easy lift to port the CoPilot CLI into Nova. 

https://githubnext.com/projects/copilot-cli/

**Reply by Halopend** [reply]
I’m guessing you mean using cli commands within the extension to essentially build a communication bridge between the code editor and the terminal but I don’t see anything particularly useful in the context of code editing within that cli interface. What we really need is an api for copilot. 

It really seems like they would rather do integration in house (which is understandable as it gives them more control to iterate without consideration of outside people complaining about api’s changing too fast) but it puts more pressure on them as various features grow out.

**Comment by c4b4d4** [comment]
+1 for Nova

**Comment by codetheorychris** [comment]
+1 for Nova

**Comment by n83b** [comment]
+1 for Nova please

**Comment by andreiwow2** [comment]
+1 for nova!

**Comment by SebastianSzturo** [comment]
https://github.com/gobijan/nova-copilot-lsp

**Reply by gobijan** [reply]
I started this last week. It's far from perfect for now. But a start for sure. Nova could use an API for previewing like sublime, vscode etc. have.

**Comment by classicrob** [comment]
Currently have the copilot extension installed, but for some reason clicking "sign in" in the extensions menu doesn't seem to be doing anything. Anyone else have this problem?

**Reply by TheBoatyMcBoatFace** [reply]
I went on a bender today and figured out a fix. I was able to get it working, but it took a few extra steps. Copilot in Nova is pretty nifty. 

I'm working on a branch now to update the docs and README on the extension page. If work isn't too chaotic tomorrow, I may be able to push by Friday. If not, I'll push early next week.

**Reply by michaelaflores** [reply]
Any updates here? This would be very useful.

**Reply by pietro-marrone-dka** [reply]
Im having the same issue

**Reply by ryndshn** [reply]
the sign in code is copied to your clipboard (I was also very confused by this but found this related issue) https://github.com/gobijan/nova-copilot-lsp/issues/5#issuecomment-1562122239

**Comment by paulgiorgi** [comment]
+1 nova

**Comment by michaelaflores** [comment]
Any movement on this?

**Reply by SageBrushes** [reply]
it's upsetting because it makes me feel like I landed on the wrong IDE

**Reply by TheBoatyMcBoatFace** [reply]
@SageBrushes I've been involved with [this plugin](https://extensions.panic.com/extensions/screenisland/screenisland.novacopilotlsp/). GitHub is being super protective of their API.

**Reply by ryndshn** [reply]
I've tried using this extension and tbh I can't even tell if it's doing anything. with the extension logs open it seems like it doesn't populate competitions. I've considered trying to build one myself but it seems daunting. Nova is the best, I hope we have a solution soon.

thanks to everyone working on this btw, it's tough.
